ion.js

```diff
@@ -70,4 +94,79 @@ class Ion {
           new Error(`Ion._handleExternalMessage - unexpected message type ${message.type}`));
     }
   }
+
+  /**
+   * Validate the provided encryption keys.
+   *
+   * @param {String} keyId
+   *        The id of the key used to encrypt outgoing data.
+   * @param {Object} key
+   *        The JSON Web Key (JWK) used to encrypt the outgoing data.
+   *        See the RFC 7517 https://tools.ietf.org/html/rfc7517
+   *        for additional information. For example:
+   *
+   *        {
+   *          "kty":"EC",
+   *          "crv":"P-256",
+   *          "x":"f83OJ3D2xF1Bg8vub9tLe1gHMzV76e8Tus9uPHvRVEU",
+   *          "y":"x_FEzRu9m36HLN_tue659LNpXW6pCyStikYjKIWI5a0",
+   *          "kid":"Public key used in JWS spec Appendix A.3 example"
+   *        }
+   *
+   * @throws {Error} if either the key id or the JWK key object are
+   *         invalid.
+   */
+  _validateEncryptionKey(keyId, key) {
+    if (typeof keyId !== "string") {
+      throw new Error(`Ion._validateEncryptionKey - Invalid encryption key id ${keyId}`);
+    }
+
+    if (typeof key !== "object") {
+      throw new Error(`Ion._validateEncryptionKey - Invalid encryption key ${key}`);
+    }
+  }
+
+  /**
+   * Submit an encrypted ping through the Ion Core addon.
+   *
+   * @param {String} payloadType
+   *        The type of the encrypted payload. This will define the
+   *        `schemaName` of the ping.
+   * @param {Object} payload
+   *        A JSON-serializable payload to be sent with the ping.
+   */
+  async sendPing(payloadType, payload) {
```

Comment:
Is this unused right now? Wondering if we should have the `background.js` pass the debug key and call `ion.sendPing()` as part of this patch so it does something useful on install (or startup, or whenever we want to send a ping)

Reply:
Yes, this is unused. I manually tested this and I'd rather focus this PR on just adding the foundations :) We don't have a clear understanding yet of what data studies need to send out of the box.

Reply:
OK thanks! sgtm to me then.